frontend/src/model/beta/Releases.tsx

```diff
@@ -26,21 +32,27 @@ export default function Releases({ model }: { model: any }) {
             Draft new Release
           </Button>
         </Box>
+        {isReleasesLoading && <Typography sx={{ m: 4 }}>{`Fetching releases for model ${model.name}`}</Typography>}
+
+        {releases && releases.length === 0 && <EmptyBlob text={`No releases found for model ${model.name}`} />}
 
-        {model.releases.map((release) => {
-          return (
-            release.name && (
-              <ModelReleaseDisplay
-                key={release.semver}
-                modelId={model.id}
-                release={release}
-                latestRelease={latestRelease}
-              />
-            )
-          )
-        })}
+        {releases &&
```

Comment:
If you make releases default to an empty array you can simplify this logic by removing the initial `releases &&`